### DISCUSSION TITLE
replace-node-text logic

### DISCUSSION THREAD

**Discussion by ryanbr** [uBlockOrigin/uAssets#24143] [Ideas]
Discussion regarding the sedCount issue, From https://github.com/brave/brave-browser/issues/39116#issuecomment-2174408736

> After some investigation, it appears that the script tag in the DOM that was injected to execute the scriptlet logic is being replaced by the scriptlet logic. i.e. the scriptlet is replacing itself.

Sample of the suggested fix;

```
     let sedCount = extraArgs.sedCount || 0;
     const handleNode = node => {
+        if (node === document.currentScript) {
+            return true;
+        }
         const before = node.textContent;
         reCondition.lastIndex = 0;
```

(or) formatted as:
```
        if (node === document.currentScript) { return true; }
```

**Comment by gorhill** [comment]
https://github.com/gorhill/uBlock/commit/cb0f65e035e4cd40b0127079ff1fca76ee964fa0

I put the test in a different place as it will always be false when executed from the mutation observer.

**Comment by ryanbr** [comment]
Is fixed in `Brave Ad Block Resources Library - Version: 1.0.91`
